devicetypes/smartthings/zigbee-multi-button.src/zigbee-multi-button.groovy

```diff
@@ -0,0 +1,226 @@
+	def map = zigbee.getEvent(description)
+	def result = map ? map : parseAttrMessage(description)
+	log.debug "Description ${description} parsed to ${result}"
+	return result
+}
+
+def parseAttrMessage(description) {
+	def descMap = zigbee.parseDescriptionAsMap(description)
+	def map = [:]
+	if (descMap?.clusterInt == zigbee.POWER_CONFIGURATION_CLUSTER && descMap.commandInt != 0x07 && descMap?.value) {
+		map = getBatteryPercentageResult(Integer.parseInt(descMap.value, 16))
+	} else if (descMap?.clusterInt == zigbee.ONOFF_CLUSTER && descMap.isClusterSpecific) {
+		map = getButtonEvent(descMap)
+	}
+	map
+}
+
+def getButtonEvent(descMap) {
+	if (descMap.commandInt == 1) {
+		getButtonResult("press")
+	}
+	else if (descMap.commandInt == 0) {
+		def button = buttonMap[device.getDataValue("model")][descMap.sourceEndpoint]
+		getButtonResult("release", button)
+	}
+}
+
+def getButtonResult(buttonState, buttonNumber = 1) {
+	def event = [:]
+	if (buttonState == 'release') {
+		def timeDiff = now() - state.pressTime
+		if (timeDiff > 10000) {
+			return event
+		} else {
+			buttonState = timeDiff < holdTime ? "pushed" : "held"
+			def descriptionText = "${device.displayName[0..-2]} button $buttonNumber was $buttonState"
+			event = createEvent(name: "button", value: buttonState, data: [buttonNumber: buttonNumber], descriptionText: descriptionText, isStateChange: true)
+			if(buttonNumber != 1) {
+				sendEventToChild(buttonNumber, event)
+				return createEvent(descriptionText: descriptionText)
+			} else {
+				return event
+			}
+		}
+	} else if (buttonState == 'press') {
+		state.pressTime = now()
+		return event
+	}
+}
+
+def sendEventToChild(buttonNumber, event) {
+	String childDni = "${device.deviceNetworkId}:$buttonNumber"
+	def child = childDevices.find { it.deviceNetworkId == childDni }
+	child?.sendEvent(event)
+}
+
+def getBatteryPercentageResult(rawValue) {
+	log.debug 'Battery'
+	def volts = rawValue / 10
+	if (volts > 3.0 || volts == 0 || rawValue == 0xFF) {
+		[:]
+	} else {
+		def result = [
+				name: 'battery'
+		]
+		def minVolts = 2.1
+		def maxVolts = 3.0
+		def pct = (volts - minVolts) / (maxVolts - minVolts)
+		result.value = Math.min(100, (int)(pct * 100))
+		def linkText = getLinkText(device)
+		result.descriptionText = "${linkText} battery was ${result.value}%"
+		createEvent(result)
+	}
+}
+
+def refresh() {
+	return zigbee.readAttribute(zigbee.POWER_CONFIGURATION_CLUSTER, batteryVoltage) +
+			zigbee.enrollResponse()
+}
+
+def ping() {
+	refresh()
+}
+
+def configure() {
+	def bindings = getModelBindings(device.getDataValue("model"))
+	return zigbee.onOffConfig() +
+			zigbee.configureReporting(zigbee.POWER_CONFIGURATION_CLUSTER, batteryVoltage, DataType.UINT8, 30, 21600, 0x01) +
+			zigbee.enrollResponse() +
+			zigbee.readAttribute(zigbee.POWER_CONFIGURATION_CLUSTER, batteryVoltage) + bindings
+}
+
+def installed() {
+	initialize()
+	sendEvent(name: "button", value: "pushed", isStateChange: true)
+	if(childDevices) {
```

Comment:
this should probably be moved to initialize, and you should probably do a `runIn(2, "initialize", [overwrite: true])` here and in updated so you don't call it twice